src/isomorphic/classic/types/__tests__/ReactPropTypesProduction-test.js

```diff
@@ -0,0 +1,220 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+describe('ReactPropTypesProduction', function() {
+  var oldProcess;
+
+  var PropTypes;
+  var React;
+  var ReactPropTypeLocations;
+  var ReactTestUtils;
+
+  beforeEach(function() {
+    __DEV__ = false;
+    oldProcess = process;
+    global.process = {env: {NODE_ENV: 'production'}};
+
+    jest.resetModuleRegistry();
+    PropTypes = require('ReactPropTypes');
+    React = require('React');
+    ReactPropTypeLocations = require('ReactPropTypeLocations');
+    ReactTestUtils = require('ReactTestUtils');
+  });
+
+  afterEach(function() {
+    __DEV__ = true;
+    global.process = oldProcess;
+  });
+
+  function expectThrowsInProduction(declaration, value) {
+    var props = {testProp: value};
+    expect(() => {
+      declaration(
+        props,
+        'testProp',
+        'testComponent',
+        ReactPropTypeLocations.prop
+      );
+    }).toThrow(
+      /React.PropTypes type checking code is stripped in production./
```

Comment:
why use a regex here and not just a string? also if it's a regex you'd probably want to escape the dot